### PR TITLE
Implement oc set route-backend and A/B to describers / printers

### PR DESCRIPTION
**Pull request by smarterclayton**
oc set route-backends foo a=10 b=20 c=0

Adds describer and printer support for router.

```
$ oc set route-backends --all
NAME         KIND     TO    WEIGHT
routes/bar   Service  bar   100 (100%)
routes/bar2  Service  bar2  100 (100%)
routes/foo   Service  a1    0
routes/foo   Service  b2    0
routes/foo   Service  c3    0

$ oc set route-backends foo a=10 b=30
route "foo" updated

$ oc set route-backends foo
NAME        KIND     TO  WEIGHT
routes/foo  Service  a   10 (25%)
routes/foo  Service  b   30 (75%)

$ oc describe route foo
Name:			foo
Namespace:		default
Created:		4 minutes ago
Labels:			<none>
Annotations:		openshift.io/host.generated=true
Requested Host:		foo-default.router.default.svc.cluster.local
Path:			<none>
TLS Termination:	edge
Insecure Policy:	<none>
Endpoint Port:		8080

Service:	a
Weight:		10 (25%)
Endpoints:	<error: endpoints "a" not found>

Service:	b
Weight:		30 (75%)
Endpoints:	<error: endpoints "b" not found>

$ oc get route
NAME      HOST/PORT                                       PATH      SERVICES        PORT      TERMINATION
bar       bar-default.router.default.svc.cluster.local              bar(100%)       9090      edge
bar2      bar2-default.router.default.svc.cluster.local             bar2(100%)      9090      edge
foo       foo-default.router.default.svc.cluster.local              a(25%),b(75%)   8080      edge
```

Primary use cases for set route-backend:

1. Allow someone to easily set a number of backends at a time: `oc set route-backends foo a=N b=N c=N`
2. Allow someone to script tweaking a particular backend up: `oc set route-backends foo --adjust a=+10%`
3. Allow someone to reset to even: `oc set route-backends foo --equal`
4. Allow someone to zero out the weights: `oc set route-backends foo --zero`

### CODE REVIEW

**Comment by smarterclayton** [comment]
[test]

**Comment by smarterclayton** [comment]
[merge] @jwforres / @fabianofranz if you can do a retroactive - we'll consider this experimental / new

File: pkg/cmd/cli/cmd/set/routebackends.go [hunk 1]
```diff
@@ -0,0 +1,585 @@
...
+
+	if o.PrintTable && o.PrintObject == nil {
+		return o.printBackends(infos)
+	}
+
+	patches := CalculatePatches(infos, o.Encoder, func(info *resource.Info) (bool, error) {
+		return UpdateBackendsForObject(info.Object, o.Transform.Apply)
+	})
+	if singular && len(patches) == 0 {
+		return fmt.Errorf("%s/%s is not a deployment config or build config", infos[0].Mapping.Resource, infos[0].Name)
```
Comment (kargakis) [inline pkg/cmd/cli/cmd/set/routebackends.go]: cp error?

Comment (smarterclayton) [inline pkg/cmd/cli/cmd/set/routebackends.go]: Yeah

**Comment by smarterclayton** [comment]
Follow up

* warn if service doesn't exist
* error if user provides percentages to "set" and they don't equal 100%
* --allow-headers doesn't work

**Pull request merged**